File: CSharp/ExtensibleCommands/ExtensibleCommandsUnitTests/TestLogger.cs
using System;
using System.IO;
using ExtensibleCommands;

namespace ExtensibleCommandsUnitTest
{
    /// <summary>
    /// Implements logger: print out in Console window
    /// </summary>
    public class TestLogger : ILog
    {
        /// <summary>
        /// Log message
        /// </summary>
        /// <param name="timestamp">Timestamp</param>
        /// <param name="level">Log level</param>
        /// <param name="message">Message to log</param>
        public void Log(string timestamp, string level, string message)
        {
            Console.WriteLine("{0} - {1} - {2}", timestamp, level, message);
        }
    }
}

File: Java/ExtensibleCommands/src/test/java/org/extensiblecommands/TestLogger.java
package org.extensiblecommands;

/**
 * Implements logger: print out in Console window
 */
public class TestLogger implements Log {
    /**
     * Log message
     * @param timestamp         Timestamp
     * @param level             Log level
     * @param message           Message to log
     */
    public void log(String timestamp, String level, String message) {
        System.out.println(String.format("%s - %s - %s", timestamp, level, message));
    }
}
